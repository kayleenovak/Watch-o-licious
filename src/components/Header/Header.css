.header {
  display: flex;
  align-items: center;
}

.heading {
  font-family: 'Shadows Into Light Two', cursive;
  color: #A6d4d3;
  font-size: 45px;
  margin: 10px 50px 0px 0px;
}

.cupcake {
  height: 65px;
  margin: 10px 0px 0px 10px;
}

.header-btn {
  box-shadow: 0px 1px 0px 0px #1c1b18;
  background: linear-gradient(to bottom, #aea7b9 5%, #74707a 100%);
  background-color: #aea7b9;
  border-radius: 10px;
  border: 2px solid #292929;
  display: inline-block;
  cursor: pointer;
  color: #ffffff;
  font-family: Arial;
  font-size: 14px;
  font-weight: bold;
  padding: 8px 16px;
  text-decoration: none;
  text-shadow: 1px 1px 0px black;
  margin: 20px 5px 0 15px;
}

.header-btn:hover {
  background: linear-gradient(to bottom, #74707a 5%, #aea7b9 100%);
  background-color: #74707a;
}

.header-btn:active {
  position: relative;
  top: 1px;
}

.active {
  background: #442c3e;
  border-radius:10px;
  border:2px solid #442c3e;
  display:inline-block;
  cursor:pointer;
  color:#ffffff;
  font-family:Arial;
  font-size:14px;
  font-weight:bold;
  padding:8px 16px;
  text-decoration:none;
  text-shadow:0px 1px 1px #442C3E;
}
