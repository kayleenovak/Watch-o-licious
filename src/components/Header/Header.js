import React, { Component } from 'react'
import { NavLink } from 'react-router-dom'
import './Header.css'

export default class Header extends Component {
  render() {
    return (
      <div className='header'>
        <img src='/cupcake.svg' className='cupcake' />
        <h1 className='heading'>Watch-o-licious</h1>
        <section className='buttons'>
          <NavLink className='header-btn' to='/favorites'>Favorites</NavLink>
          <NavLink className='header-btn' to='/watched'>Watched</NavLink>
          <NavLink className='header-btn' to='/watchlist'>Watch List</NavLink>
          <NavLink className='header-btn' exact to='/'>Home</NavLink>
        </section>
      </div>
    )
  }
}