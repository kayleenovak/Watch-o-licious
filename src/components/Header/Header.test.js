import React from 'react'
import Header from './Header.js'
import { shallow } from 'enzyme'
import { NavLink } from 'react-router-dom'

describe('Header', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Header />)
  })

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  })

  it('should render 3 buttons with 3 NavLinks', () => {

    expect(wrapper.find('button').length).toEqual(4)
    expect(wrapper.find(NavLink).length).toEqual(4)
  })
})