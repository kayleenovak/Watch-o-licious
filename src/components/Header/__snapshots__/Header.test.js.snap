// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="cupcake"
          src="/cupcake.svg"
        />,
        <h1
          className="heading"
        >
          Watch-o-licious
        </h1>,
        <section
          className="buttons"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="header-btn"
            to="/favorites"
          >
            Favorites
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="header-btn"
            to="/watched"
          >
            Watched
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="header-btn"
            to="/watchlist"
          >
            Watch List
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="header-btn"
            exact={true}
            to="/"
          >
            Home
          </NavLink>
        </section>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "cupcake",
          "src": "/cupcake.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Watch-o-licious",
          "className": "heading",
        },
        "ref": null,
        "rendered": "Watch-o-licious",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="header-btn"
              to="/favorites"
            >
              Favorites
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="header-btn"
              to="/watched"
            >
              Watched
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="header-btn"
              to="/watchlist"
            >
              Watch List
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="header-btn"
              exact={true}
              to="/"
            >
              Home
            </NavLink>,
          ],
          "className": "buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Favorites",
              "className": "header-btn",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Watched",
              "className": "header-btn",
              "to": "/watched",
            },
            "ref": null,
            "rendered": "Watched",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Watch List",
              "className": "header-btn",
              "to": "/watchlist",
            },
            "ref": null,
            "rendered": "Watch List",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Home",
              "className": "header-btn",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="cupcake"
            src="/cupcake.svg"
          />,
          <h1
            className="heading"
          >
            Watch-o-licious
          </h1>,
          <section
            className="buttons"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="header-btn"
              to="/favorites"
            >
              Favorites
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="header-btn"
              to="/watched"
            >
              Watched
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="header-btn"
              to="/watchlist"
            >
              Watch List
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="header-btn"
              exact={true}
              to="/"
            >
              Home
            </NavLink>
          </section>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "cupcake",
            "src": "/cupcake.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Watch-o-licious",
            "className": "heading",
          },
          "ref": null,
          "rendered": "Watch-o-licious",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="header-btn"
                to="/favorites"
              >
                Favorites
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="header-btn"
                to="/watched"
              >
                Watched
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="header-btn"
                to="/watchlist"
              >
                Watch List
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="header-btn"
                exact={true}
                to="/"
              >
                Home
              </NavLink>,
            ],
            "className": "buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Favorites",
                "className": "header-btn",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Watched",
                "className": "header-btn",
                "to": "/watched",
              },
              "ref": null,
              "rendered": "Watched",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Watch List",
                "className": "header-btn",
                "to": "/watchlist",
              },
              "ref": null,
              "rendered": "Watch List",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Home",
                "className": "header-btn",
                "exact": true,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
